### ISSUE TITLE
Fix S1067 FP: Don't report in If directive trivia condition

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6574]
### Description

S1067 reports in `#if` compiler directives. The syntax is limited in the conditions expressions allowed and nesting of `#if` is also discouraged. S1067 should not report in `#if` and `#elif` condition expression context.

```cs
#if NETCF_1_0
#elif NETCF_2_0 || NET_2_0 || MONO_2_0 || MONO_3_5 || MONO_4_0
//    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Noncompliant FP
#endif
```
see e.g. [peach AYUVHDDKlZ5lVLD55SRL (internal only)](https://peach.sonarsource.com/project/issues?issues=AYUVHDDKlZ5lVLD55SRL&open=AYUVHDDKlZ5lVLD55SRL&id=dnn)

* C#/VB.NET Plugins version: Unreleased current master at 5736bd38f67211c3ca37639559cb9d5f6b362c5b

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1792](https://sonarsource.atlassian.net/browse/NET-1792)

[NET-1792]: https://sonarsource.atlassian.net/browse/NET-1792?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
